src/pages/Swap/index.tsx

```diff
@@ -410,134 +108,14 @@ export default function Swap() {
-                    </SwitchIconContainer>
-                    <CurrencyInputPanel
-                      value={formattedAmounts[Field.OUTPUT]}
-                      onUserInput={handleTypeOutput}
-                      onMax={handleMaxInput(Field.OUTPUT)}
-                      currency={currencies[Field.OUTPUT]}
-                      onCurrencySelect={handleOutputSelect}
-                      otherCurrency={currencies[Field.INPUT]}
-                      fiatValue={fiatValueOutput}
-                      priceImpact={priceImpact}
-                      isFallbackFiatValue={isFallbackFiatValueOutput}
-                      maxAmount={maxAmountOutput}
-                      showCommonBases
-                      disabled={isInputPanelDisabled}
-                      id="swap-currency-output"
-                    />
-                  </AutoColumn>
-
-                  <TradeDetails
-                    show={!showWrap}
-                    loading={loading}
-                    trade={trade}
-                    bestPricedTrade={bestPricedTrade}
-                    showAdvancedSwapDetails={showAdvancedSwapDetails}
-                    setShowAdvancedSwapDetails={setShowAdvancedSwapDetails}
-                    recipient={recipient}
-                  />
-                  <SwapButtons
-                    wrapInputError={wrapInputError}
-                    showApproveFlow={showApproveFlow}
-                    userHasSpecifiedInputOutput={userHasSpecifiedInputOutput}
-                    approval={approval}
-                    setSwapState={setSwapState}
-                    priceImpactSeverity={priceImpactSeverity}
-                    swapCallbackError={swapCallbackError}
-                    wrapType={wrapType}
-                    approvalSubmitted={approvalsSubmitted[currentTradeIndex]}
-                    currencies={currencies}
-                    trade={trade}
-                    swapInputError={swapInputError}
-                    swapErrorMessage={swapErrorMessage}
-                    loading={loading}
-                    onWrap={onWrap}
-                    approveCallback={approveCallback}
-                    handleSwap={handleSwap}
-                    handleInputSelect={handleInputSelect}
-                    wrapState={wrapState}
-                    setWrapState={setWrapState}
-                  />
-                </AutoColumn>
-              </Wrapper>
-            </AppBody>
-            {showAdvancedSwapDetails && (
-              <AdvancedSwapDetailsDropdown
-                isLoading={loading}
-                trade={trade}
-                allPlatformTrades={allPlatformTrades}
-                onSelectedPlatformChange={setPlatformOverride}
+              <ChartOptionsButtons
+                hasBothCurrenciesInput={hasBothCurrenciesInput}
+                selectedChartOption={selectedChartOption}
+                setselectedChartOption={setselectedChartOption}
```

Comment:
We are only using it here, do you think it's worth it?

Reply:
Since it's not being used elsewhere in `Swap/index` I think we can extract. Code will be cleaner in this component. But it's just a small improvement.

Reply:
Ok, fair, I'll do that.